### DISCUSSION TITLE
Why is juicefs listening on ports 6070, 6060, 9567 ?

### DISCUSSION THREAD

**Discussion by cayolblake** [juicedata/juicefs#320] [Q&A] [answered]
Hello,

Would like to know why is juicefs listening on ports 6060, 6070, 9567? 

One of them `9567` is listening on all interfaces.

I remember reading somewhere that one of them is related to a web UI. 

Is there a way to disable these?

**Comment by cayolblake** [comment, marked as answer]
So, I was able to dirty patch `cmd/mount.go` to comment parts of code related to metrics HTTP server `9567` and the `gops` agent binding to ports `6070` and `6060` and rebuild the `juicefs` binary however I guess there should be a much easier way to do that.

**Reply by xiaogaozi** [reply]
`6060` is [pprof](https://golang.org/pkg/net/http/pprof) HTTP server, `6070` is [gops](https://github.com/google/gops) agent, `9567` is Prometheus metrics.

May I ask, why you want to disable these ports?

**Reply by cayolblake** [reply]
Well, primarily for security reasons, more open ports means more attack vectors to worry about, especially if one of the ports is binding to all interfaces publicly by default.

Secondary being an on-needed basis would give better control to users and sysadmins to enable/disable as required :)

**Reply by xiaogaozi** [reply]
Thanks for your suggestion, we'll consider add options to disable these ports.

**Reply by xiaogaozi** [reply]
https://github.com/juicedata/juicefs/issues/325 FYI, I've created an issue.
